es-core/src/renderers/Renderer.h

```diff
@@ -67,6 +81,7 @@ namespace Renderer
 	void        pushClipRect    (const Vector2i& _pos, const Vector2i& _size);
 	void        popClipRect     ();
 	void        drawRect        (const float _x, const float _y, const float _w, const float _h, const unsigned int _color, const unsigned int _colorEnd, bool horizontalGradient = false, const Blend::Factor _srcBlendFactor = Blend::SRC_ALPHA, const Blend::Factor _dstBlendFactor = Blend::ONE_MINUS_SRC_ALPHA);
+	void        drawRect        (const float _x, const float _y, const float _w, const float _h, Gradient::Mode _mode, const unsigned int _color, const unsigned int _colorEnd, const Blend::Factor _srcBlendFactor = Blend::SRC_ALPHA, const Blend::Factor _dstBlendFactor = Blend::ONE_MINUS_SRC_ALPHA);
```

Comment:
Any reason why the argument order was changed? In the existing call, the gradient parameter came after the colors, but in this new method it comes before.  It would be good to have consistency here.  It would make it easier to eventually migrate away from the old method.

I would have expected something along the lines of the following, with gradient mode after the colors, and a default value for gradient mode matching the behavior of the current `drawRect`:  

```c++
void drawRect(const float _x, const float _y, const float _w, const float _h, const unsigned int _color, const unsigned int _colorEnd, Gradient::Mode _mode = Gradient::Mode::TO_BOTTOM, const Blend::Factor _srcBlendFactor = Blend::SRC_ALPHA, const Blend::Factor _dstBlendFactor = Blend::ONE_MINUS_SRC_ALPHA);
```

Reply:
Can be considered, if the gradient parameter is made mandatory for both methods. Otherwise autocasting can have undesired results.